File: src/indigo-ui/src/models/Invoice.ts
export interface Invoice {
    InvoiceId: number;
    Description: string;
    BankAccount: BankAccount;
    Bank: Bank;
    Vendor: Vendor;
    Status: InvoiceStatus;
    InvoiceNumber: string;
    InvoiceDate: string;
    DueDate: string;
    Amount: number;
    IsDeleted: boolean;
}

export enum InvoiceStatus {
    Pending = 1,
    Paid = 2,
    Overdue = 3,
    Void = 4
}

interface BankAccount {
    BankId: number;
    BankAccountId: number;
    AccountNumber: string;
    AccountName: string;
}

interface Bank {
    BankId: number;
    BankName: string;
}

interface Vendor {
    VendorId: number;
    VendorName: string;
}

File: src/indigo-services/indigo-invoicing-service/Indigo.Invoicing.Domain/Models/Invoice.cs
using Indigo.Invoicing.Domain.Entities;

namespace Indigo.Invoicing.Domain.Models
{
    public class Invoice
    {
        public int InvoiceId { get; set; }
        public string? InvoiceNumber { get; set; }
        public string? Description { get; set; }
        public DateTime InvoiceDate { get; set; }
        public DateTime DueDate { get; set; }
        public float Amount { get; set; }
        public bool IsDeleted { get; set; }
        public Status Status { get; set; }
        public BankAccount BankAccount { get; set; } = new BankAccount();
        public Bank Bank { get; set; } = new Bank();
    }
}
